internal/const.go

```diff
@@ -17,3 +17,12 @@ var LegacyHelloLowercase = "ismaster"
 
 // LegacyNotPrimary is the legacy version of the "not primary" server error message.
 var LegacyNotPrimary = "not master"
+
+// MinSupportedWire is the minimum wire version supported by the driver.
+const MinSupportedWire = 6
+
+// MaxSupportedWire is the maximum wire version supported by the driver.
+const MaxSupportedWire = 17
+
+// MinSupportedMongoDB is the version string for the lowest MongoDB version supported by the driver.
+const MinSupportedMongoDB = "3.6"
```

Comment:
Optional: Since these constants are only used in the `topology` package (in `fsm.go`), consider moving them there.

Reply:
@matthewdale I'm kind of conflicted on this. Intuitively it seems like we'd want to be able to access version constraints from anywhere in the app without worrying about a dependency cycle. But maybe putting them in topology satisfies this?

Reply:
I generally prefer keeping code localized to where it's used unless it's shared between multiple uses. I could see an argument for moving `SupportedWireVersions` to `internal/const.go` since it's used in 2 different packages.

As far as constants vs literals for `SupportedWireVersions`, I'd recommend using literals so `SupportedWireVersions.Min` and `SupportedWireVersions.Max` remain the single source of truth for wire version support.

Reply:
Sounds good, to avoid cluttering things up since this is used in a single case I'll revert all of the internal/const.go logic.